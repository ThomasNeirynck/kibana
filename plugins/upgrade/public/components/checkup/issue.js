import PropTypes from 'prop-types';
import React from 'react';

import { IssueSymbol } from './issue_symbol';


export function Issue({ details, level, message, url }) {
  return (
    <li className='kuiMenuItem'>
      <div className='kuiEvent'>
        <IssueSymbol level={ level } />

        <div className='kuiEventBody'>
          <div className='kuiEventBody__message'>
            { message }{ details ? `: ${details }` : null}
          </div>
          <div className='kuiEventBody__metadata'>
            {
              url
              ? <a className='kuiLink' href={ url }>View Documentation</a>
              : null
            }
          </div>
        </div>
      </div>
    </li>
  );
}

Issue.propTypes = {
  details: PropTypes.string,
  level: PropTypes.string,
  message: PropTypes.string,
  url: PropTypes.string,
};

Issue.defaultProps = {
  details: null,
  level: null,
  message: null,
  url: null,
};
