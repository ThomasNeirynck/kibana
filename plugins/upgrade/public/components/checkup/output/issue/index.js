import React from 'react';

import { IssueSymbol } from '../issue_symbol';


export function Issue({ details, level, message, url }) {
  return (
    <li className='kuiMenuItem'>
      <div className='kuiEvent'>
        <IssueSymbol level={ level } />

        <div className='kuiEventBody'>
          <div className='kuiEventBody__message'>
            { message }{ details ? `: ${details }` : null}
          </div>
          <div className='kuiEventBody__metadata'>
            {
              url
              ? <a className='kuiLink' href={ url }>View Documentation</a>
              : null
            }
          </div>
        </div>
      </div>
    </li>
  );
}

Issue.propTypes = {
  details: React.PropTypes.string,
  level: React.PropTypes.string,
  message: React.PropTypes.string,
  url: React.PropTypes.string,
};

Issue.defaultProps = {
  details: null,
  level: null,
  message: null,
  url: null,
};
