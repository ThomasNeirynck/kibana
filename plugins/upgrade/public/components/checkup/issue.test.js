import { shallow } from 'enzyme';
import React from 'react';

import { Issue } from './issue';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('Issue', () => {
  test('renders the details property', () => {
    const component = (
      <Issue details="issue details" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders the level property', () => {
    const component = (
      <Issue level="info" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders the message property', () => {
    const component = (
      <Issue message="a message" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders the url property', () => {
    const component = (
      <Issue url="/some/url" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
