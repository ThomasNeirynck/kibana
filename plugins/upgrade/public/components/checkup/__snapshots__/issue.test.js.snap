// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue renders the details property 1`] = `
<li
  className="kuiMenuItem"
>
  <div
    className="kuiEvent"
  >
    <IssueSymbol
      level={null}
    />
    <div
      className="kuiEventBody"
    >
      <div
        className="kuiEventBody__message"
      >
        : issue details
      </div>
      <div
        className="kuiEventBody__metadata"
      />
    </div>
  </div>
</li>
`;

exports[`Issue renders the level property 1`] = `
<li
  className="kuiMenuItem"
>
  <div
    className="kuiEvent"
  >
    <IssueSymbol
      level="info"
    />
    <div
      className="kuiEventBody"
    >
      <div
        className="kuiEventBody__message"
      />
      <div
        className="kuiEventBody__metadata"
      />
    </div>
  </div>
</li>
`;

exports[`Issue renders the message property 1`] = `
<li
  className="kuiMenuItem"
>
  <div
    className="kuiEvent"
  >
    <IssueSymbol
      level={null}
    />
    <div
      className="kuiEventBody"
    >
      <div
        className="kuiEventBody__message"
      >
        a message
      </div>
      <div
        className="kuiEventBody__metadata"
      />
    </div>
  </div>
</li>
`;

exports[`Issue renders the url property 1`] = `
<li
  className="kuiMenuItem"
>
  <div
    className="kuiEvent"
  >
    <IssueSymbol
      level={null}
    />
    <div
      className="kuiEventBody"
    >
      <div
        className="kuiEventBody__message"
      />
      <div
        className="kuiEventBody__metadata"
      >
        <a
          className="kuiLink"
          href="/some/url"
        >
          View Documentation
        </a>
      </div>
    </div>
  </div>
</li>
`;
